satellite/satellitedb/overlaycache.go

```diff
@@ -33,6 +33,71 @@ type overlaycache struct {
 	db *satelliteDB
 }
 
+// SelectAllStorageNodesUpload returns all nodes that qualify to store data, organized as reputable nodes and new nodes
+func (cache *overlaycache) SelectAllStorageNodesUpload(ctx context.Context, selectionCfg overlay.NodeSelectionConfig) (reputable, new []*overlay.SelectedNode, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	if selectionCfg.MinimumVersion == "" {
```

Comment:
As I mentioned, changing which nodes are selected should be a separate PR either before or after this PR.

However, changing this would require changing quite a lot of code due to storj-sim, backwards compatibility tests etc.